### PR TITLE
Upstream master bump 0513

### PR DESCRIPTION
**Pull request by jjsjann123**
Updating nvfuser code base.

This should fix the indexing issue observed in https://github.com/pytorch/vision/issues/6015.

Running tests locally as well. Will update the description here at a later point

@bypass-github-export-checks

### CODE REVIEW

**Comment by jjsjann123** [comment]
cc'ing @davidberard98 
Don't think we can get nightlies to run on this. I'll mark it as ready for review once my local tests finished.
Likely would get some style changes as well. :crossed_fingers:

File: aten/src/ATen/core/interned_strings.h
Comment (jjsjann123) [inline aten/src/ATen/core/interned_strings.h:54]: This should be the only change that's outside of nvfuser code base.

**Comment by davidberard98** [comment]
Marking as "ready for review" so I can import it.

**Comment by malfet** [comment]
Just curious: considering that commits are cheap in GHF world: is there really a benefit in such aggregate commits?
Obvious drawbacks: 
- Regression from one particular improvements results in revert of the entire change
- it's hard for community developers to contribute to NVFuser
- Large PRs are harder to keep uptodate with trunk
- It's hard to populate changelog
- Individual contributions are lost in this bulk commit

**Comment by seemethere** [comment]
> Just curious: considering that commits are cheap in GHF world: is there really a benefit in such aggregate commits? Obvious drawbacks:
> 
> * Regression from one particular improvements results in revert of the entire change
> * it's hard for community developers to contribute to NVFuser
> * Large PRs are harder to keep uptodate with trunk
> * It's hard to populate changelog
> * Individual contributions are lost in this bulk commit

+1 to this, I see there are commits dating all the way back to December in this pull request.

Maintaining a branch this big must be difficult to do, is there not a way we can make this easier for you to contribute these changes without maintaining such a large divergence?

**Review by seemethere — CHANGES_REQUESTED** [review]
Considering we have a branch cut for 1.12 coming soon and a PR of this size could introduce a lot of uncertainty for having a stable release branch I say we should hold off on merging this PR until after the branch cut occurs and additional testing can be done within nightlies.

**Comment by davidberard98** [comment]
@seemethere these are patches required to re-enable nvfuser by default, which is planned for 1.12.

**Comment by seemethere** [comment]
> @seemethere these are patches required to re-enable nvfuser by default, which is planned for 1.12.
> 
> cc @atalman to verify that this is okay

We probably won't even have time to verify that these changes are okay in the nightlies prior to the branch cut though so I don't see how we can plan to enable it by default in that time frame.

**Comment by davidberard98** [comment]
@jjsjann123 is it possible to get a smaller PR to fix to the torchvision issue instead

**Comment by davidberard98** [comment]
@seemethere are you suggesting we land this after 5/19 and then cherry-pick into 1.12 if nightlies are okay, or to postpone it entirely?

**Comment by eellison** [comment]
@seemethere I understand this is a big pr to cherry-pick, but we're still ahead of the branch cut. Imo it's also less risky because there is a well-defined fallback path - switch default enablement to NNC.

**Comment by jjsjann123** [comment]
> @jjsjann123 is it possible to get a smaller PR to fix to the torchvision issue instead

This PR targets more than just patching the torchvision issue. Perf tuning on kernels in this devel update is also an important target for our 1.12 enablement. :pray:

**Comment by seemethere** [comment]
> @seemethere are you suggesting we land this after 5/19 and then cherry-pick into 1.12 if nightlies are okay, or to postpone it entirely?

I'd say _any_ PR >10k LOC shouldn't be merged into master so close to a branch cut or cherry picked into a release branch. This doesn't just apply to this particular PR.

**Comment by malfet** [comment]
@eellison please do not use githubbot to merge this change but rather let it go thru internal CI/CD to get the full signal.

**Review by eellison — APPROVED** [review]

**Review by seemethere — APPROVED** [review]

**Comment by eellison** [comment]
> Merge failed due to list index out of range
Raised by https://github.com/pytorch/pytorch/actions/runs/2347398630

cc @seemethere, do you know anything about this ? it landed internally

**Comment by eellison** [comment]
```
  File ".github/scripts/trymerge.py", line 923, in main
    pr.merge_into(repo, dry_run=args.dry_run, force=args.force, comment_id=args.comment_id)
  File ".github/scripts/trymerge.py", line 695, in merge_into
    repo._run_git("commit", f"--author=\"{self.get_author()}\"", "-m", msg)
  File ".github/scripts/trymerge.py", line 571, in get_author
    authors = self.get_authors()
  File ".github/scripts/trymerge.py", line 566, in get_authors
    rc[self.get_committer_login(idx)] = self.get_committer_author(idx)
  File ".github/scripts/trymerge.py", line 527, in get_committer_author
    return self._fetch_authors()[num][1]
```

**Comment by seemethere** [comment]
Looks like this PR is so massive that our GHF tooling can't actually handle it correctly, investigating on how to unblock

**Comment by seemethere** [comment]
Merged manually with the following modifications to `trymerge.py`:

```diff
diff --git a/.github/scripts/trymerge.py b/.github/scripts/trymerge.py
index f482358a85..067ca14184 100755
--- a/.github/scripts/trymerge.py
+++ b/.github/scripts/trymerge.py
@@ -521,10 +521,12 @@ class GitHubPR:
         return authors

     def get_committer_login(self, num: int = 0) -> str:
+        return "jjsjann123"
         return self._fetch_authors()[num][0]

     def get_committer_author(self, num: int = 0) -> str:
-        return self._fetch_authors()[num][1]
+        return "jjsjann123 <jiej@nvidia.com>"
+        # return self._fetch_authors()[num][1]

     def get_checkrun_conclusions(self) -> Dict[str, str]:
         """ Returns list of checkrun / conclusions """
```

**Comment by seemethere** [comment]
This is the _last_ time we will be merging a PR of 20k lines

**Pull request closed without merging**
